p4-fusion/commands/change_list.cc

```diff
@@ -6,19 +6,19 @@
  */
 #include "change_list.h"
 
+#include <utility>
+
 #include "p4_api.h"
 #include "describe_result.h"
 #include "filelog_result.h"
 #include "print_result.h"
 #include "utils/std_helpers.h"
 #include "minitrace.h"
 
-#include "thread_pool.h"
-
-ChangeList::ChangeList(const std::string& clNumber, const std::string& clDescription, const std::string& userID, const int64_t& clTimestamp)
+ChangeList::ChangeList(const int& clNumber, std::string clDescription, std::string userID, const int64_t& clTimestamp)
     : number(clNumber)
-    , user(userID)
-    , description(clDescription)
+    , user(std::move(userID))
```

Comment:
`std::move` is not about compiler optimizations. Specifically, the `std::string` constructor has multiple overloads:
- A copy constructor - this takes a `const std::string &` as an argument.
- A move constructor - this takes a `std::string &&` as an argument.

Based on how overload resolution works, if you try to initialize a field of type `std::string` with an r-value of type `std::string &` (which is implicitly created), it will prefer the copy constructor by adding `const`. Using `std::move` converts the `std::string &` to a `std::string &&` and hence the move constructor will be used.